plugin-nabu/src/main/java/ca/on/oicr/gsi/shesmu/nabu/NabuPlugin.java

```diff
@@ -0,0 +1,155 @@
+package ca.on.oicr.gsi.shesmu.nabu;
+
+import ca.on.oicr.gsi.shesmu.plugin.Definer;
+import ca.on.oicr.gsi.shesmu.plugin.cache.ReplacingRecord;
+import ca.on.oicr.gsi.shesmu.plugin.cache.ValueCache;
+import ca.on.oicr.gsi.shesmu.plugin.input.ShesmuInputSource;
+import ca.on.oicr.gsi.shesmu.plugin.json.JsonListBodyHandler;
+import ca.on.oicr.gsi.shesmu.plugin.json.JsonPluginFile;
+import ca.on.oicr.gsi.status.SectionRenderer;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.nio.file.Path;
+import java.time.Instant;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+public class NabuPlugin extends JsonPluginFile<NabuConfiguration> {
+
+  private final class CaseArchiveCache
+      extends ValueCache<Stream<NabuCaseArchiveValue>, Stream<NabuCaseArchiveValue>> {
+
+    private CaseArchiveCache(Path fileName) {
+      super("case_archive " + fileName.toString(), 30, ReplacingRecord::new);
+    }
+
+    private Stream<NabuCaseArchiveValue> caseArchives(String baseUrl)
+        throws IOException, InterruptedException {
+      return HTTP_CLIENT
+          .send(
+              HttpRequest.newBuilder(URI.create(baseUrl + "/cases")).GET().build(),
+              new JsonListBodyHandler<>(MAPPER, NabuCaseArchiveDto.class))
+          .body()
+          .get()
+          .map(
+              ca -> {
+                return new NabuCaseArchiveValue(
+                    ca.getCaseFilesUnloaded() == null
+                        ? Optional.empty()
+                        : Optional.of(Instant.parse(ca.getCaseFilesUnloaded())),
+                    ca.getCaseIdentifier(),
+                    Optional.ofNullable(ca.getCommvaultBackupJobId()),
+                    Instant.parse(ca.getCreated()),
+                    ca.getFilesCopiedToOffsiteArchiveStagingDir() == null
+                        ? Optional.empty()
+                        : Optional.of(Instant.parse(ca.getFilesCopiedToOffsiteArchiveStagingDir())),
+                    ca.getFilesLoadedIntoVidarrArchival() == null
+                        ? Optional.empty()
+                        : Optional.of(Instant.parse(ca.getFilesLoadedIntoVidarrArchival())),
+                    ca.getLimsIds(),
+                    Instant.parse(ca.getModified()),
+                    ca.getRequisitionId(),
+                    ca.getWorkflowRunIdsForOffsiteArchive(),
+                    ca.getWorkflowRunIdsForVidarrArchival() == null
+                        ? Optional.empty()
+                        : Optional.of(ca.getWorkflowRunIdsForVidarrArchival()));
+              });
+    }
+
+    @Override
+    protected Stream<NabuCaseArchiveValue> fetch(Instant lastUpdated) throws Exception {
+      if (config.isEmpty()) {
+        return Stream.empty();
```

Comment:
please `return new ErrorableStream<>(Stream.empty(), false);` in these cases

Reply:
done